File: 1110.py
while True:
  n = int(input())
  if(n == 0): break

  cartas = [i for i in range(1,n+1)]
  descart = []

  while(len(cartas) > 1):
    descart.append(cartas.pop(0))
    cartas.append(cartas.pop(0))

  print("Discarded cards: ", end="")
  for i in range(len(descart) - 1):
    print(str(descart[i])+", ", end="")
  print(descart[-1])
  print("Remaining card:", cartas[0])

File: 1110.cpp
#include <iostream>

using namespace std;

void print(int *a, int n)
{
  for (int i = 0; i < n; i++)
    cout << a[i] << " ";
  cout << endl;
}

int main()
{
  int n;

  while (true)
  {
    cin >> n;
    if (n <= 0)
      break;

    int remaining[100];
    for (int i = 0; i < n; i++)
      remaining[i] = i + 1;

    int i = 0;
    int size = n;

    cout << "Discarded cards: ";
    while (size > 1)
    {
      if (size != 2)
        cout << remaining[i] << ", ";
      else
        cout << remaining[i] << endl;

      i++;
      size--;
      remaining[n++] = remaining[i];
      i++;
    }
    cout << "Remaining card: " << remaining[i] << endl;
  }
  return 0;
}